Add Scope description and informative appendix on possible metadata usage (#9)
peppelinux/federation-wallet: openid-federation-wallet-1_0.md

this document are to be interpreted as described in BCP 14 [@!RFC2119]
[@!RFC8174] when, and only when, they appear in all capitals, as shown here.

# Scope

This specification is a profile of [@!OpenID.Federation] for wallet ecosystems.
It defines Entity Types for entities participating in those ecosystems.

> It describes trust evaluation for those entities.

Reviewer: ```suggestion
It describes technical trust evaluation mechanisms for those entities.
```

|                          Trust Anchor                        |
+--------------------------------------------------------------+
````
**Figure 1**: The relationships and interactions within a Wallet ecosystem using the Four-Party Model

> The Figure above illustrates at the center the Holder, who interacts directly with both the Credential Issuer and the Relying Party. The Credential Issuer provides Digital Credentials to the Holder, while the Relying Party relies on these Credentials to verify the Holder's claims. Above the Holder is the Wallet Provider, which facilitates the registration and the attestation of the security and integrity of the Holder. All entities, including the Credential Issuer, Holder, and Relying Party, are underpinned by a Trust Anchor, which provides a foundational layer of trust and security for the entire system. This setup ensures that all interactions and transactions are anchored in a trusted framework.

Reviewer: ```suggestion
The Figure above illustrates at the center the Holder, who interacts directly with both the Credential Issuer and the Relying Party. The Credential Issuer provides Digital Credentials to the Holder, while the Relying Party relies on these Credentials to verify the Holder's claims. Above the Holder is the Wallet Provider, which facilitates the registration and the attestation of the security and integrity of the Holder. All entities, including the Credential Issuer, Relying Party, Wallet Provider and therefore Holders, and are underpinned by a Trust Anchor, which provides a foundational layer of trust and security for the entire system. This setup ensures that all interactions and transactions are anchored in a trusted framework.
```

Reviewer: this resolves https://github.com/peppelinux/federation-wallet/issues/6

TBD: concept of RP instance in the form of verifier app in control of a natural person.

> # Wallet Entity Types

Reviewer: ```suggestion
# Wallet Architecture Entity Types
```

Reviewer: I have decided to include the word Architecture to better describe the scope of this section. The entities types defined therein are not related to wallet or wallet instance but the entire wallet architecture

| Relying Party         | `federation_entity`, `openid_wallet_relying_party`         | [@!OpenID.Federation], [@!OpenID4VP], this specification       |

The Credential Issuer is an OAuth 2.0 Protected Resource Server and it MAY also implement, within the same Entity, an OAuth 2.0 Authorization Server. According to [@!OpenID4VCI], the Authorization Server can be external to the Entity that implements the Credential Endpoint, therefore the use of `oauth_authorization_server` is OPTIONAL.

## OpenID Wallet Provider Entity Type

The OpenID Federation Entity Type Identifier for the Wallet Provider is `openid_wallet_provider`.

For information on metadata parameters specific to OpenID Wallets,
refer to Section *8. Wallet Metadata (Authorization Server Metadata)* of
the OpenID for Verifiable Presentations [@!OpenID4VP] specification.

## OpenID Credential Issuer Entity Type

The OpenID Federation Entity Type Identifier for the Credential Issuer is `openid_credential_issuer`.

For information on metadata parameters specific to OpenID Credential Issuers,
refer to Section *10.2. Credential Issuer Metadata* of
the OpenID for Verifiable Credential Issuance [@!OpenID4VCI] specification.

> ## OpenID Wallet Relying Party Entity Type

Reviewer: this is for a brand new issue:

since we have the name credential issuer, and therefore the enntity type identifier openid_credential_issuer, for sake of normalization and omogenity, it would be convenient to rename openid_wallet_relying_party to openid_credential_verifier

@selfissued WDYT?

Author: Yes, let's do this.  I'll create a PR after merging this one.

Reviewer: I have open a brand new issue for this, therefore this comment doesn't block the merge of this PR

  </front>
</reference>

# Possible Use of Metadata Parameters by Wallet Entities

This non-normative appendix speculates about metadata parameters that might be used
by the Entity Types defined by this specification.
Some of these are already defined by other specifications.
Some are yet to be defined, and may be defined in the future by other specifications.
Some are likely to change.

> Some may end up not being used at all.

Reviewer: ```suggestion
```

Please read this illustrative content with these caveats in mind.

## Possible OpenID Wallet Provider Metadata Usage

Possible metadata parameters that could be used by
OpenID Wallet Providers with the `openid_wallet_provider` Entity Type Identifier
are listed in the table below.

The table and examples below use metadata parameters defined by
[@!OpenID.Federation], [@!RFC8414], and [@!OpenID4VP].

| **Metadata Parameter** | **Availability** | **Value** |
|----------|----------- | ------|
| jwks | REQUIRED | A JSON Web Key Set (JWKS) that represents the  Wallet Provider's public keys. |

> | token_endpoint | REQUIRED | Endpoint for obtaining the Wallet Instance Attestation. |

Reviewer: ```suggestion
| authorization_endpoint | REQUIRED | Endpoint for obtaining the authorization for the issuance of the Wallet Attestation. |
| token_endpoint | REQUIRED | Endpoint for obtaining the Wallet Attestation. |
```

| aal_values_supported | OPTIONAL | List of supported values for the Authenticator Assurance Level, as defined in [ NIST Special Publication 800-63B](https://pages.nist.gov/800-63-3/sp800-63b.html#sec4). These values specify the security level of the app. Values are trust framework specific. |
| grant_types_supported | REQUIRED | The grant types supported by the token endpoint.|
| token_endpoint_auth_methods_supported | REQUIRED | Supported authentication methods for the token endpoint.|
| token_endpoint_auth_signing_alg_values_supported          | REQUIRED | Supported signature algorithms for the token endpoint.|

Below is a non-normative example of `openid_wallet_provider` metadata:

````
{

>   "token_endpoint": "https://wallet.provider.example/token",

Reviewer: ```suggestion
  "authorization_endpoint": "https://wallet.provider.example/authorization",
  "token_endpoint": "https://wallet.provider.example/token",
```
